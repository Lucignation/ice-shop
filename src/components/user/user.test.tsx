import User from './user.component';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

const mockUser = {
  id: 1,
  email: 'lucignation@gmail.com',
  username: 'lucignation',
  password: 'password',
  name: {
    firstname: 'Gerald',
    lastname: 'Olumide',
  },
  address: {
    city: 'Lagos',
    street: 'my street',
    number: 45,
    zipcode: '434',
    geolocation: {
      lat: '-43556',
      long: '4688',
    },
  },
  phone: '345676543456',
};

const mockProps = {
  user: mockUser,
};

describe('User ', () => {
  it('renders a snapshot of <User /> when it it passed into a component ', () => {
    const user = renderer
      .create(
        <Router>
          <User {...mockProps} />
        </Router>
      )
      .toJSON();

    expect(user).toMatchSnapshot();
  });
});
