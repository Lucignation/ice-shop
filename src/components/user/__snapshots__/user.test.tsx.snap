// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User  renders a snapshot of <User /> when it it passed into a component  1`] = `
<div
  className="card user_card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      <span
        className="title"
      >
        Full Name
      </span>
      : 
      Gerald
       
      Olumide
    </h5>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
       
      <span
        className="title"
      >
        Username
      </span>
      : 
      lucignation
    </h6>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
      <span
        className="title"
      >
        Address
      </span>
      : 
      45
      ,
       
      my street
      , 
      Lagos
      , 
      434
    </h6>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
       
      <span
        className="title"
      >
        Phone
      </span>
      : 
      345676543456
    </h6>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
       
      <span
        className="title"
      >
        Email
      </span>
      : 
      lucignation@gmail.com
    </h6>
    <h6
      className="card-subtitle mb-2 text-muted"
    >
      <span
        className="title"
      >
        Lat
      </span>
      : 
      -43556
       
      <span
        className="title"
      >
        Long
      </span>
      : 
      4688
    </h6>
    <a
      className="btn btn-info"
      href="/user/1"
      onClick={[Function]}
    >
      Edit
    </a>
  </div>
</div>
`;
